config/bootstrap.php

```diff
@@ -22,6 +22,13 @@
     class_alias('Cake\Utility\Text', 'Cake\Utility\String');
 }
 
+// @deprecated Backward compatibility with 2.x, 3.0.x
+class_alias('Cake\Mailer\AbstractTransport', 'Cake\Network\Email\AbstractTransport');
+class_alias('Cake\Mailer\Transport\DebugTransport', 'Cake\Network\Email\DebugTransport');
+class_alias('Cake\Mailer\Email', 'Cake\Network\Email\Email');
+class_alias('Cake\Mailer\Transport\MailTransport', 'Cake\Network\Email\MailTransport');
+class_alias('Cake\Mailer\Transport\SmtpTransport', 'Cake\Network\Email\SmtpTransport');
```

Comment:
I personally prefer @lorenzo's suggestion of moving this stuff to the app template, however, worries me for already created apps which would need to manually update or having breaking changes to deal with. 

If we want, as @ceeram suggested, we can pass a 3rd param to control autoloading but why aren't we doing it for the `Text` class as well?